client/src/components/MeetingExperience.test.tsx

```diff
@@ -161,74 +161,96 @@ describe('MeetingExperience', () => {
     expect(callWithChatComposites.length).toBe(1);
   });
 
-  it('should render Survey component when postcall is defined and valid', async () => {
-    const setRenderPostCallMock = jest.fn();
-    const useStateMock: any = (_: any) => [true, setRenderPostCallMock];
-    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+  // it('should render Survey component when postcall is defined and valid', async () => {
```

Comment:
We should at least bring this test back: `should not render Survey component when postcall is undefined`. Then we can separately address the other case and add a test later for the case where postcall is defined and valid.

Reply:
Setting useState to true makes callWithChatApapdter get set to "true" too (since it is a state value) . if i remember right from my trial and error, the test will break in MeetingExperience.tsx where callWithChatAdapter is accessed to get the callID. testing state is considered fragile.

Reply:
https://stackoverflow.com/questions/57692521/how-to-mock-the-usestate-hook-implementation-so-that-it-actually-changes-the-sta

Reply:
I wrote some test that mocks `@azure/communication-react` package that we can control (mock) props to `MeetingExperience` component to test.

We no longer need to mock `useState` anymore since it is bad practice and it is not possible to test component that has more than one state.

Please take a look [this file](https://github.com/Azure-Samples/communication-services-virtual-visits-js/blob/4e6415473b7ad13926ec274d6d3038b9622362e5/client/src/components/MeetingExperience.test.tsx)

Reply:
that worked! nice find, please add any reference links you have on how you landed on this approach. I made some tweaks to also include callWithChatComposite check when survey is not rendered